src/engine/SCons/Node/Python.py

```diff
@@ -177,7 +184,7 @@ def get_csig(self, calc=None):
         return contents
 
 
-def ValueWithMemo(value, built_value=None):
+def ValueWithMemo(value, built_value=None, name=None):
     global _memo_lookup_map
```

Comment:
you need to pass name to the created Value().

Reply:
Ah sorry, I missed that. I have added a test that was broken with the previous code and is fixed now.